### DISCUSSION TITLE
Client connection to RabbitMQ deployed on OpenShift

### DISCUSSION THREAD

**Discussion by andrescolodreroarion** [rabbitmq/cluster-operator#1371] [General]
Hi,
After create the rabbitmq cluster with the operator, i wonder if there is a way to connect to  RabbitMQ via https route, or do i need to create nodeport/externals IP for every deployed cluster?

**Comment by Zerpet** [comment]
I'm far from being an expert in OpenShift, but it looks like you can create a Route [1] to access HTTP/HTTPs traffic (among others). This limits the exposure to only RabbitMQ management UI/API. You should be able to configure a route to select the service that exports the management UI; IIRC, the service that exposes RMQ protocol ports has the same name as your custom resource name.

If you want to expose the AMQP, or other non-http protocol, you probably need a different ingress provider that supports TCP ingress; or expose those protocol ports as `NodePort` or `LoadBalancer` type services.

[1] https://docs.openshift.com/container-platform/3.11/architecture/networking/routes.html

**Comment by andrescolodrero** [comment]
HI,
i ended testing this with MetalLB and creating an AddressPool to the rabbbitMQ service.
